tests/storage/blocksd_test.py

```diff
@@ -862,6 +863,43 @@ def test_volume_metadata(tmp_storage, tmp_repo, fake_access, fake_rescan,
     assert data == md.storage_format(domain_version, CAP=md.capacity)
 
 
+@requires_root
+@pytest.mark.root
+@pytest.mark.xfail(reason='Vdsm over-allocates one chunk', strict=False)
+@pytest.mark.parametrize("capacity", [
+    # capacity
+    1 * MiB,
+    512 * MiB,
+    sc.VG_EXTENT_SIZE,
+    CHUNK_SIZE_MB * MiB - sc.VG_EXTENT_SIZE,
+    (CHUNK_SIZE_MB + 1) * MiB - sc.VG_EXTENT_SIZE,
+    (CHUNK_SIZE_MB - 1) * MiB,
```

Comment:
Not sure why we need so many tests. The interesting cases are:
- capacity == chunk size -> allocate capacity
- capacity == chunk size - 1 -> allocate capacity
- capacity < extent size -> allocate extent size
- capacity > chunk size -> allocated chunk (but this must be already covered by other tests)

(Allocations are always rounded up to extent size)

These tests are slow, so minimizing the number of tests is very important. Using parameters for this tests is really bad - we will create the loop devices and storage domain for creating a single volume for every test!

All the cases should be tested with the same storage domain. You can create one test that create 3 volumes instead of using parameters.

Reply:
Ok, I did put the three usecases together, avoided using parametrized test, and commented the whole test to help to read it.